cmds/core/dhclient/dhclient_test.go

```diff
@@ -5,36 +5,52 @@
 package main
 
 import (
+	"bytes"
+	"log"
 	"strings"
 	"testing"
 
-	"github.com/u-root/u-root/pkg/testutil"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv6"
 )
 
 var tests = []struct {
-	iface  string
-	isIPv4 string
+	iface  []string
+	isIPv4 bool
 	out    string
+	err    string
 }{
 	{
-		iface:  "nosuchanimal",
-		isIPv4: "-ipv4=true",
-		out:    "no interfaces match nosuchanimal\n",
+		iface:  []string{"nosuchanimal"},
+		isIPv4: true,
+		out:    "",
+		err:    "no interfaces match nosuchanimal",
+	},
+	{
+		iface:  []string{"nosuchanimal1", "nosuchanimaltoo"},
+		isIPv4: true,
+		out:    "",
+		err:    "more than one interface specified",
 	},
 }
 
 func TestDhclient(t *testing.T) {
+	var out = bytes.NewBuffer(nil)
+	log.SetOutput(out)
 	for _, tt := range tests {
-		out, err := testutil.Command(t, tt.isIPv4, tt.iface).CombinedOutput()
-		if err == nil {
-			t.Errorf("%v: got nil, want err", tt)
+		out.Reset()
+		err := run(tt.iface, 15, 5, false, false, false, tt.isIPv4, true, dhcpv6.DefaultServerPort, "ff02::1:2", dhcpv4.ServerPort)
+
+		if err != nil && tt.err == "" {
+			t.Errorf("no error expected, got: \n%v", err)
+		} else if err == nil && tt.err != "" {
+			t.Errorf("error \n%v\nexpected, got nil error", tt.err)
+		} else if err != nil && err.Error() != tt.err {
+			t.Errorf("error \n%v\nexpected, got: \n%v", tt.err, err)
 		}
-		if !strings.HasSuffix(string(out), tt.out) {
-			t.Errorf("expected:\n%s\ngot:\n%s", tt.out, string(out))
+
+		if !strings.HasSuffix(out.String(), tt.out) {
```

Comment:
(here and in other similar code in other test files) If `err` (from line 42) is not nil then you can't rely on `out.String()` to be accurate so these test outputs may be incorrect. It might be best to skip this test code if an error was returned by `run`

Reply:
Good catch, I fixed this in 6209b6a as well.